store/tikv/gcworker/gc_worker.go

```diff
@@ -640,105 +648,109 @@ func (w *GCWorker) checkUseDistributedGC() (bool, error) {
+	runner.SetMaxBackoff(tikv.GcResolveLockMaxBackoff)
+	err := runner.RunOnRange(ctx, []byte(""), []byte(""))
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	logutil.Logger(ctx).Info("[gc worker] finish resolve locks",
+		zap.String("uuid", w.uuid),
+		zap.Uint64("safePoint", safePoint),
+		zap.Int32("regions", runner.CompletedRegions()),
+		zap.Duration("cost time", time.Since(startTime)))
+	metrics.GCHistogram.WithLabelValues("resolve_locks").Observe(time.Since(startTime).Seconds())
+	return nil
+}
+
+func (w *GCWorker) resolveLocksForRegion(
+	ctx context.Context,
+	bo *tikv.Backoffer,
+	startKey []byte,
+	loc *tikv.KeyLocation,
+	safePoint uint64,
+) ([]byte, error) {
 	req := &tikvrpc.Request{
 		Type: tikvrpc.CmdScanLock,
 		ScanLock: &kvrpcpb.ScanLockRequest{
 			MaxVersion: safePoint,
 			Limit:      gcScanLockLimit,
+			StartKey:   startKey,
 		},
 	}
 
-	logutil.Logger(ctx).Info("[gc worker] start resolve locks",
-		zap.String("uuid", w.uuid),
-		zap.Uint64("safePoint", safePoint))
-	startTime := time.Now()
-	regions, totalResolvedLocks := 0, 0
-
-	var key []byte
-	for {
-		select {
-		case <-ctx.Done():
-			return errors.New("[gc worker] gc job canceled")
-		default:
-		}
-
-		bo := tikv.NewBackoffer(ctx, tikv.GcResolveLockMaxBackoff)
+	select {
+	case <-ctx.Done():
+		return nil, errors.New("[gc worker] gc job canceled")
+	default:
+	}
 
-		req.ScanLock.StartKey = key
-		loc, err := w.store.GetRegionCache().LocateKey(bo, key)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		resp, err := w.store.SendReq(bo, req, loc.Region, tikv.ReadTimeoutMedium)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		regionErr, err := resp.GetRegionError()
+	resp, err := w.store.SendReq(bo, req, loc.Region, tikv.ReadTimeoutMedium)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	regionErr, err := resp.GetRegionError()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if regionErr != nil {
+		err = bo.Backoff(tikv.BoRegionMiss, errors.New(regionErr.String()))
 		if err != nil {
-			return errors.Trace(err)
-		}
-		if regionErr != nil {
-			err = bo.Backoff(tikv.BoRegionMiss, errors.New(regionErr.String()))
-			if err != nil {
-				return errors.Trace(err)
-			}
-			continue
-		}
-		locksResp := resp.ScanLock
-		if locksResp == nil {
-			return errors.Trace(tikv.ErrBodyMissing)
-		}
-		if locksResp.GetError() != nil {
-			return errors.Errorf("unexpected scanlock error: %s", locksResp)
-		}
-		locksInfo := locksResp.GetLocks()
-		locks := make([]*tikv.Lock, len(locksInfo))
-		for i := range locksInfo {
-			locks[i] = tikv.NewLock(locksInfo[i])
+			return nil, errors.Trace(err)
 		}
+		return startKey, nil
+	}
+	locksResp := resp.ScanLock
+	if locksResp == nil {
+		return nil, errors.Trace(tikv.ErrBodyMissing)
+	}
+	if locksResp.GetError() != nil {
+		return nil, errors.Errorf("unexpected scanlock error: %s", locksResp)
+	}
+	locksInfo := locksResp.GetLocks()
+	locks := make([]*tikv.Lock, len(locksInfo))
```

Comment:
Don't need to resolve locks when locks is empty.

Reply:
This is the same as  the old code and `BatchResolveLocks` will skip processing if `len(locks) == 0`. I think it's ok to keep it.